WatchFaceAlphaKotlin/app/src/main/java/com/example/android/wearable/alpha/data/db/WatchFaceDatabaseCallback.kt

```diff
@@ -0,0 +1,211 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.android.wearable.alpha.data.db
+
+import android.content.Context
+import android.content.res.Resources
+import android.util.Log
+import androidx.room.RoomDatabase
+import androidx.sqlite.db.SupportSQLiteDatabase
+import com.example.android.wearable.alpha.R
+import kotlin.random.Random
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+// All constants are used for populating the database only.
+private const val CENTER_CIRCLE_DIAMETER_FRACTION = 0.03738f
+private const val OUTER_CIRCLE_STROKE_WIDTH_FRACTION = 0.00467f
+private const val NUMBER_STYLE_OUTER_CIRCLE_RADIUS_FRACTION = 0.00584f
+
+private const val GAP_BETWEEN_OUTER_CIRCLE_AND_BORDER_FRACTION = 0.03738f
+private const val GAP_BETWEEN_HAND_AND_CENTER_FRACTION =
+    0.01869f + CENTER_CIRCLE_DIAMETER_FRACTION / 2.0f
+
+private const val NUMBER_RADIUS_FRACTION = 0.45f
+
+private const val HOUR_HAND_LENGTH_FRACTION = 0.21028f
+private const val HOUR_HAND_WIDTH_FRACTION = 0.02336f
+
+private const val MINUTE_HAND_LENGTH_FRACTION = 0.3783f
+private const val MINUTE_HAND_WIDTH_FRACTION = 0.0163f
+
+private const val SECOND_HAND_LENGTH_FRACTION = 0.37383f
+private const val SECOND_HAND_WIDTH_FRACTION = 0.00934f
+
+// Primary key ids assigned for various style and dimension elements in the database.
+// These are accessible to the watch face, so it can assign the element the user chooses in
+// the watch face settings.
+const val hourHandDimensionsKeyId = "hour_key_id"
+const val minuteHandDimensionsKeyId = "minute_key_id"
+const val secondHandDimensionsKeyId = "second_key_id"
+
+const val ambientColorStyleKeyId = "ambient_style_key_id"
+const val redColorStyleKeyId = "red_style_key_id"
+const val greenColorStyleKeyId = "green_style_key_id"
+const val blueColorStyleKeyId = "blue_style_key_id"
+const val whiteColorStyleKeyId = "white_style_key_id"
+
+private const val TAG = "WatchFaceDatabaseCallback"
+
+/**
+ * Callback used when the database is created. It's used in our case to populate the database on
+ * first use with all the information you need to render a watch face.
+ */
+class WatchFaceDatabaseCallback(
+    private val context: Context,
+    private val scope: CoroutineScope
+) : RoomDatabase.Callback() {
+    override fun onCreate(db: SupportSQLiteDatabase) {
+        super.onCreate(db)
+        WatchFaceDatabase.getDatabase(context, scope).let { database ->
```

Comment:
To use the Room DAO, I need an instance of the DB, and I can't get that until the .build() is called ([code line](https://github.com/android/wear-os-samples/blob/real-alpha/WatchFaceAlphaKotlin/app/src/main/java/com/example/android/wearable/alpha/data/db/WatchFaceDatabase.kt#L53)).

So, I'm accessing it via the static method (assuming it has been initialized since addCallback() is called afterwards).

I'm doing it from that method allows it to be initialized as well. I revised the code to assign the INSTANCE right away to help.

I have a followup PR with some minor data layer changes, so if that doesn't cover it, just let me know and I can resolve there for you.

Reply:
I also need the context in a followup PR to populate the DB, i.e., to use the strings, dimensions, drawable ids as we discussed earlier, but feel free to let me know changes there and I can change.